### PR TITLE
Add aarch64 support

### PR DESCRIPTION
**Pull request by ananos**
- add aarch64 platform for rumprun
- update src-netbsd (custom fixes needed for aarch64 & rumprun)

*Stack protection in NetBSD is off (pending further investigation)

Please note that buildrump.sh & src-netbsd both point to forked repos.

### CODE REVIEW

File: build-rr.sh [hunk 1]
```diff
@@ -379,7 +381,14 @@ buildrump ()
...
 	[ "${MACHINE_GNU_ARCH}" = "x86_64" ] \
-	    && extracflags="-F CFLAGS=-mno-red-zone"
+	    && extracflags='-F CFLAGS=-mno-red-zone'
+
+	# Disable new errors on GCC 7 which break netbsd-src compilation
+	#
+	[ `${CC} -dumpversion | cut -f1 -d.` -ge 7 ] \
+		&& extracflags="$extracflags -F CPPFLAGS=-Wimplicit-fallthrough=0"
+
+
```
Comment (ricarkol) [inline build-rr.sh:391]: extra line

File: lib/libbmk_rumpuser/rumpuser_synch.c [hunk 2]
```diff
@@ -211,6 +210,12 @@ rumpuser_mutex_owner(struct rumpuser_mtx *mtx, struct lwp **lp)
 	*lp = mtx->o;
 }
 
+int
+rumpuser_mutex_spin_p(struct rumpuser_mtx *mtx)
```
Comment (ricarkol) [inline lib/libbmk_rumpuser/rumpuser_synch.c:214]: is this one used by the new netbsd? can't find the caller

File: build-rr.sh [hunk 3]
```diff
@@ -542,6 +551,14 @@ makeconfig ()
 	else
 		echo "CONFIG_CXX=no" >> ${1}
 	fi
+
+	# Check for if compiler supports -no-pie and save to EXTRACCFLAGS
+	gccnopie=
```
Comment (ricarkol) [inline build-rr.sh:556]: can you explain this a bit please?

File: platform/solo5/arch/evbarm64-el/machdep.c [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+#include <bmk-core/printf.h>
+#include <bmk-core/sched.h>
+unsigned long bmk_cpu_arm_curtcb;
+
+
+void
+bmk_platform_cpu_sched_settls(struct bmk_tcb *next)
+{
+        __asm__ __volatile__("msr tpidr_el0, %0" :: "r"(next->btcb_tp));
```
Comment (ricarkol) [inline platform/solo5/arch/evbarm64-el/machdep.c:36]: This is not supposed to be called when not using TLS (in hvt and spt). You could assert that (to catch bugs).

File: .travis.yml [hunk 5]
```diff
@@ -4,22 +4,30 @@ language: c
...
 
 before_script:
   - sudo apt-get update -y
   - sudo apt-get install qemu-kvm libxen-dev libseccomp-dev bats genisoimage sudo iproute2 wget -y
   - sudo apt-get install --only-upgrade binutils gcc -y
+  - bash travis_wait.sh 60 git submodule update --init
 
 env:
   - PLATFORM=solo5 MACHINE=x86_64 TESTS=spt EXTRAFLAGS=
-  - PLATFORM=hw MACHINE=x86_64 TESTS=qemu EXTRAFLAGS=
```
Comment (ricarkol) [inline .travis.yml:15]: is this because the tests fail? ideally, it would be nice to keep them

File: .travis.yml [hunk 6]
```diff
@@ -4,22 +4,30 @@ language: c
...
-  - PLATFORM=hw MACHINE=i486 ELF=elf TESTS=qemu EXTRAFLAGS='-- -F ACLFLAGS=-m32 -F ACLFLAGS=-march=i686'
-  - PLATFORM=hw MACHINE=x86_64 TESTS=qemu EXTRAFLAGS= CXX='false'
   - PLATFORM=hw MACHINE=x86_64 TESTS=none KERNONLY=-k EXTRAFLAGS= 
 
 script:
-  - git submodule update --init
-  - ./build-rr.sh -o myobj -j16 -qq ${KERNONLY} ${PLATFORM} ${EXTRAFLAGS}
+  - export CC=gcc-6
```
Comment (ricarkol) [inline .travis.yml:29]: I think i asked this before, but the new netbsd can only be built with gcc 6 (or higher) right?

**Review by ricarkol — CHANGES_REQUESTED** [review]
Looks good. Just have some questions and would prefer to re-enable the qemu tests.

File: lib/libbmk_rumpuser/rumpuser_synch.c (hunk 2 above)
Comment (papazof) [reply lib/libbmk_rumpuser/rumpuser_synch.c:214]: yeap, you can find the declaration here: https://github.com/cloudkernels/src-netbsd/blob/b84d52a97472825058b398b561b2263dc556b70c/sys/rump/include/rump/rumpuser.h#L197
and it's used here: https://github.com/cloudkernels/src-netbsd/blob/b84d52a97472825058b398b561b2263dc556b70c/sys/rump/librump/rumpkern/locks.c#L164

File: platform/solo5/arch/evbarm64-el/machdep.c (hunk 4 above)
Comment (papazof) [reply platform/solo5/arch/evbarm64-el/machdep.c:36]: OK, sure.

File: build-rr.sh (hunk 3 above)
Comment (papazof) [reply build-rr.sh:556]: aah, this is a patch merged in upstream rumprun, if you see the following lines it checks if gcc is provided with "-pie" as default (as is the case for many debian versions) and sets -no-pie to solve build issues with these compilers.

File: .travis.yml (hunk 5 above)
Comment (papazof) [reply .travis.yml:15]: Yeap, these are failing currently.

File: .travis.yml (hunk 6 above)
Comment (papazof) [reply .travis.yml:29]: Correct for the x86 case (for aarch64 gcc-5 is fine too).

File: lib/libbmk_rumpuser/rumpuser_synch.c (hunk 2 above)
Comment (ricarkol) [reply lib/libbmk_rumpuser/rumpuser_synch.c:214]: Thanks

**Review by ricarkol — APPROVED** [review]

**Pull request merged**